### PR TITLE
cap32: add joystick button fire3

### PR DESCRIPTION
**Pull request by revvv**
Amstrad joysticks have three fire buttons. I added the missing fire3 button.

I have tested the code on my 3DS.

New button layout:
A = fire1
B = fire2
X = fire3 (previously unused)
Y = fire1 (previously mapped to space)

@DSkywalk Please verify ;-)
The key codes in `retro_events.h` are not used in a remap file, so IMHO it's ok to shift them.

### CODE REVIEW

**Comment by DSkywalk** [comment]
imagen
The atari standard usually only allows two buttons because one pin carries 5v.

Please can you tell me which controller (no hombrew) that works on an real amstrad has 3 buttons.

Thanks :bow:

**Comment by revvv** [comment]
And note that in fire3 is already in the current code for [player1](https://github.com/libretro/libretro-cap32/blob/22d0475be7f6249669358d02b6cba4bc98abe936/libretro/retro_events.h#L132) and [player2](https://github.com/libretro/libretro-cap32/blob/22d0475be7f6249669358d02b6cba4bc98abe936/libretro/retro_events.h#L142):
```
   CPC_KEY_SPARE                  = 0x96,
   ...
   CPC_KEY_JOY2_FIRE3             = 0x66, // just DIY JOYSTICK
```

I just renamed `CPC_KEY_SPARE ` to `CPC_KEY_JOY_FIRE3`
So 4 years ago you were aware of fire3 ;-)

File: libretro/retro_events.c [hunk 1]
```diff
@@ -148,15 +148,15 @@ t_button_cfg btnPAD[MAX_PADCFG] = {
    }},
    {{ // JOYSTICK PLAYER 1 CFG
    CPC_KEY_JOY_FIRE2,   // B
-   CPC_KEY_NULL,        // Y
+   CPC_KEY_JOY_FIRE1,   // Y
```
Comment (DSkywalk) [inline libretro/retro_events.c:151]: Please remove this change. These buttons are used to make binding simple, see: #63

**Review by DSkywalk — COMMENTED** [review]
I leave you these changes to test the branch, thank you @revvv :)

**Comment by DSkywalk** [comment]
> Well, if you like. There are other unbound RetroPad buttons like L, R,Start, Select etc. Off topic: 1, 2, ENTER etc could also be helpful for some games...

Please note that we have been working for a year to create configurations for the games and change the button configuration now would break many of them and they would have to be redone. This part is very critical :+1:

File: libretro/retro_events.c [hunk 2]
```diff
@@ -148,15 +148,15 @@ t_button_cfg btnPAD[MAX_PADCFG] = {
...
    CPC_KEY_NULL,        // SELECT
    CPC_KEY_NULL,        // START
    CPC_KEY_JOY_UP,      // DUP
    CPC_KEY_JOY_DOWN,    // DDOWN
    CPC_KEY_JOY_LEFT,    // DLEFT
    CPC_KEY_JOY_RIGHT,   // DRIGHT
    CPC_KEY_JOY_FIRE1,   // A
-   CPC_KEY_NULL,        // X
```
[the remark this answers is not in the document]
Comment (revvv) [reply libretro/retro_events.c]: Removed for now

**Comment by LibretroAdmin** [comment]
Hi @DSkywalk , leaving it entirely in your hands as to how to proceed. You can decide to merge PRs as you see fit for this repo

**Comment by DSkywalk** [comment]
Hello @LibretroAdmin :)

Before proceeding, I will need to check that everything is working properly. However, Thank you for all your trust in my vision and support during this time :heart:

**Comment by DSkywalk** [comment]
Tested OK :)

**Comment by revvv** [comment]
Tested nightly Windows x64 build: **OK**
Thank you!!!

**Pull request merged**